packages/fuels-signers/src/provider.rs

```diff
@@ -226,28 +247,31 @@ impl Provider {
     }
 
     /// Get transaction by id.
-    pub async fn get_transaction_by_id(&self, tx_id: &str) -> io::Result<TransactionResponse> {
+    pub async fn get_transaction_by_id(
+        &self,
+        tx_id: &str,
+    ) -> Result<TransactionResponse, ProviderError> {
         Ok(self.client.transaction(tx_id).await.unwrap().unwrap())
     }
 
     // - Get transaction(s)
     pub async fn get_transactions(
         &self,
         request: PaginationRequest<String>,
-    ) -> std::io::Result<PaginatedResult<TransactionResponse, String>> {
-        self.client.transactions(request).await
+    ) -> Result<PaginatedResult<TransactionResponse, String>, ProviderError> {
+        Ok(self.client.transactions(request).await?)
     }
 
-    // - Get transaction(s) by owner
+    /// Get transaction(s) by owner
     pub async fn get_transactions_by_owner(
         &self,
         owner: &str,
         request: PaginationRequest<String>,
-    ) -> std::io::Result<PaginatedResult<TransactionResponse, String>> {
-        self.client.transactions_by_owner(owner, request).await
+    ) -> Result<PaginatedResult<TransactionResponse, String>, ProviderError> {
+        Ok(self.client.transactions_by_owner(owner, request).await?)
```

Comment:
```suggestion
        self.client
            .transactions_by_owner(owner, request)
            .await
            .map_err(Into::into)
 ```
 
 I believe I saw you do this in a previous PR. It is more transparent but the current version seems more... ergonomic? 
 
 Dunno just wanted to point it out.